nanorl/algorithms/dpo/model.py

```diff
@@ -0,0 +1,21 @@
+import torch
+import torch.nn.functional as F
+
+
+def get_model_logprobs(model, input_ids, attention_mask, labels):
+    with torch.cuda.amp.autocast():
+        outputs = model(input_ids=input_ids, attention_mask=attention_mask)
+        logits = outputs.logits
+
+        shift_logits = logits[:, :-1, :].contiguous()
+        shift_labels = labels[:, 1:].contiguous()
+
+        log_probs = F.log_softmax(shift_logits, dim=-1)
+        gathered_log_probs = torch.gather(
+            log_probs, dim=-1, index=shift_labels.unsqueeze(-1)
+        ).squeeze(-1)
+
+        mask = (shift_labels != -100) & (shift_labels != model.config.pad_token_id)
```

Comment:
good
consider setting this 100 in the config

`token_cut_off = 100`

Reply:
It is pretty well defined in the eco-system of torch to use `-100` to ignore.

<img width="818" alt="image" src="https://github.com/user-attachments/assets/62212697-93b9-47eb-8bc9-f55bc8e1b002" />

Reply:
cool, lets keep as is